### PR TITLE
feat: check if the request is a dry-run to match CAPI topology design

### LINKED ISSUE #264: ClusterClass incompatibility caused by KubevirtMachineTemplateSpec immutability.
**What steps did you take and what happened:**
Actually, at our company, we are using the CAPI ClusterClass feature to manage our clusters lifecycle.
We saw that the capi-controller-manager tried to update KubevirtMachineTemplate when we update a field (like the memory of cpu cores) but we're facing this error :
```
"Reconciler error" err="error reconciling the Cluster topology: failed to reconcile MachineDeployment/worker-djv5f: failed to create patch helper for KubevirtMachineTemplate/worker-infra-fdjc7: server side apply dry-run failed for modified object: admission webhook \"validation.kubevirtmachinetemplate.infrastructure.cluster.x-k8s.io\" denied the request: KubevirtMachineTemplateSpec is immutable
```

Did you know why KubevirtMachineTemplate resources are immutable, is this for security, lifecycle of any other purpose ? 

**What did you expect to happen:**
The KubevirtMachineTemplate should be updated and a new machine set will be triggered by the cluster-api to rollout worker nodes. 

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by aamoyel**
Fix #264

### CODE REVIEW

File: pkg/webhookhandler/validator_test.go [hunk 1]
```diff
@@ -173,13 +173,15 @@ func newRequest(operation admissionv1.Operation, oldObj, newObj *v1alpha1.Kubevi
...
 
+	var defaultReqDryRun bool = false
 	switch operation {
 	case admissionv1.Create:
 		req.Object = runtime.RawExtension{
 			Raw:    []byte(runtime.EncodeOrDie(encoder, oldObj)),
 			Object: oldObj,
 		}
 	case admissionv1.Update:
+		req.DryRun = &defaultReqDryRun
```
Comment (nunnatsa) [inline pkg/webhookhandler/validator_test.go:184]: Please replace with `ptr.To(false)` and drop `defaultReqDryRun`

**Review by nunnatsa — COMMENTED** [review]
Thanks for the PR!

/ok-to-test

File: pkg/webhookhandler/validator_test.go (hunk 1 above)
Comment (aamoyel) [reply pkg/webhookhandler/validator_test.go]: Done, thx for the tip :)

Comment (nunnatsa) [reply pkg/webhookhandler/validator_test.go]: It failed to compile in the e2e test: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubernetes-sigs_cluster-api-provider-kubevirt/277/pull-kubernetes-sigs-cluster-api-provider-kubevirt-e2e/1764970340280176640

please run `go mod tidy`

**Comment by aamoyel** [comment]
Tests needs to be fixed @nunnatsa ? Or just re request ?

**Comment by nunnatsa** [comment]
I don't think we need to touch the test. just run `go mod tidy` it will probably modify `go.sum`, maybe also `go.mod`. just push the result.

BTW, we currently can't re-run e2e. the only way is to push again, e.g.
```
git commit --amend
git push -f
```

**Comment by aamoyel** [comment]
No changes after many e2e restart

**Comment by nunnatsa** [comment]
> No changes after many e2e restart

Found it: you'll need to also run `go mod tidy` from the `e2e` directory.

**Pull request merged**